code/parse/parselo.cpp

```diff
@@ -2763,161 +2709,132 @@ void stuff_boolean(bool *b, bool a_to_eol)
 	diag_printf("Stuffed bool: %s\n", (b) ? NOX("true") : NOX("false"));
 }
 
-int stuff_bool_list(bool *blp, int max_bools)
+//	Stuff an integer value (cast to a ubyte) pointed at by Mp.
+//	Advances past integer characters.
+void stuff_ubyte(ubyte *i)
 {
-	int count = 0;
-	bool trash_buf = false;
+	int temp;
+	stuff_int(&temp);
+	*i = (ubyte)temp;
+}
+
+template <typename T, typename F>
+void stuff_token_list(SCP_vector<T> &list, F stuff_one_token, const char *type_as_string)
+{
+	list.clear();
 
 	ignore_white_space();
 
-	if (*Mp != '(') {
-        error_display(1, "Reading boolean list.  Found [%c].  Expecting '('.\n", *Mp);
-        throw parse::ParseException("Syntax error");
+	if (*Mp != '(')
+	{
+		error_display(1, "Reading %s list.  Found [%c].  Expected '('.\n", type_as_string, *Mp);
+		throw parse::ParseException("Syntax error");
 	}
-
 	Mp++;
 
 	ignore_white_space();
 
-	while(*Mp != ')')
+	while (*Mp != ')')
 	{
-		if(count < max_bools)
-		{
-			stuff_boolean(&blp[count++], false);
-			ignore_white_space();
+		T item;
+		if (stuff_one_token(&item))
+			list.push_back(std::move(item));
 
-			//Since Bobb has set a precedent, allow commas for bool lists -WMC
-			if(*Mp == ',')
-			{
-				Mp++;
-				ignore_white_space();
-			}
-		}
-		else
-		{
-			trash_buf = true;
-			break;
-		}
-	}
+		ignore_white_space();
 
-	if(trash_buf)
-	{
-		error_display(0, "Boolean list has more than allowed arguments; max is %d. Arguments over max will be ignored.", max_bools);
-		while(*Mp != ')')
+		if (*Mp == ',')
 		{
-			stuff_boolean(&trash_buf, false);
+			Mp++;
 			ignore_white_space();
 		}
 	}
-
 	Mp++;
-
-	return count;
 }
 
-
-//	Stuff an integer value (cast to a ubyte) pointed at by Mp.
-//	Advances past integer characters.
-void stuff_ubyte(ubyte *i)
+template <typename T, typename F>
+size_t stuff_token_list(T *listp, size_t list_max, F stuff_one_token, const char *type_as_string)
 {
-	int temp;
-	stuff_int(&temp);
-	*i = (ubyte)temp;
+	SCP_vector<T> list;
+	stuff_token_list(list, stuff_one_token, type_as_string);
+
+	if (list_max < list.size())
+	{
+		error_display(0, "Too many items in %s list.  Found " SIZE_T_ARG "; max is " SIZE_T_ARG ".  List has been truncated.", type_as_string, list.size(), list_max);
+		list.resize(list_max);
+	}
+
+	size_t i = 0;
+	for (const auto &item : list)
+		listp[i++] = item;
+
+	Assert(i == list.size());
+	return i;
 }
 
-int parse_string_flag_list(int *dest, flag_def_list defs[], int defs_size)
+void parse_string_flag_list(int *dest, flag_def_list defs[], size_t defs_size)
 {
 	Assert(dest!=NULL);	//wtf?
 
-	char (*slp)[NAME_LENGTH] = (char(*)[32])new char[defs_size*NAME_LENGTH];
-	int num_strings = stuff_string_list(slp, defs_size);
-	int i, j;
+	SCP_vector<SCP_string> slp;
+	stuff_string_list(slp);
 
-	for(i = 0; i < num_strings; i++)
+	for (auto &str : slp)
 	{
-		for(j = 0; j < defs_size; j++)
+		for (size_t j = 0; j < defs_size; j++)
```

Comment:
Small nitpick: Wouldn't it make sense to rename `j` to `i` here since the previous `i` variable is gone?

Reply:
That's fair.